tests/unit/test_hyper_transformer.py

```diff
@@ -1159,6 +1175,191 @@ def test_update_default_sdtype_transformers(self):
+            - ``self._provided_field_sdtypes`` has been updated.
+            - ``self._provided_field_transformers`` has been updated.
+        """
+        # Setup
+        instance = HyperTransformer()
+        instance._fitted = False
+        instance.field_transformers = {'a': object()}
+        mock_transformer = Mock()
+        mock_transformer.get_input_type.return_value = 'datetime'
+        column_name_transformer = {
+            'my_column': mock_transformer
+        }
+
+        # Run
+        instance.update_transformers(column_name_transformer)
+
+        # Assert
+        mock_warnings.warn.assert_not_called()
+        assert instance.field_transformers['my_column'] == mock_transformer
+        assert instance.field_sdtypes['my_column'] == 'datetime'
+        assert instance._provided_field_sdtypes == {'my_column': 'datetime'}
+        assert instance._provided_field_transformers == {'my_column': mock_transformer}
+
+    @patch('rdt.hyper_transformer.print')
+    def test_update_transformers_no_field_transformers(self, mock_print):
+        """Test update transformers.
+
+        Ensure that the function updates properly the ``self.field_transformers`` and prints the
+        expected messages to guide the end-user.
+
+        Setup:
+            - Initialize ``HyperTransformer`` with ``_fitted`` as ``False``.
+            - Dictionary with a ``column_name`` and a ``Mock`` transformer.
+
+        Mock:
+            - Patch the ``print`` function in order to ensure that expected message is being
+              printed to the end user.
+            - Transformer, mock for the replacement transformer.
+
+        Side Effects:
+            - ``self.field_transformers`` has been updated.
+            - ``self.field_sdtypes`` has been updated.
+            - ``self._provided_field_sdtypes`` has been updated.
+            - ``self._provided_field_transformers`` has been updated.
+        """
+        # Setup
+        instance = HyperTransformer()
+        instance._fitted = False
+        mock_transformer = Mock()
+        mock_transformer.get_input_type.return_value = 'datetime'
+        column_name_transformer = {
+            'my_column': mock_transformer
+        }
+
+        # Run
+        instance.update_transformers(column_name_transformer)
+
+        # Assert
+        expected_message = (
+            'Tip: Use the `detect_initial_config` method to pre-populate all the sdtypes '
+            'and transformers from your dataset.'
+        )
+        mock_print.assert_called_once_with(expected_message)
+        assert instance.field_transformers['my_column'] == mock_transformer
+        assert instance.field_sdtypes['my_column'] == 'datetime'
+        assert instance._provided_field_sdtypes == {'my_column': 'datetime'}
+        assert instance._provided_field_transformers == {'my_column': mock_transformer}
+
+    @patch('rdt.hyper_transformer.print')
+    def test_update_transformers_field_transformer_exists(self, mock_warnings):
```

Comment:
minor: can we change this name to reflect that it tests if the sdtype and transformer don't match?